docs/source/tutorials/interactive_simulation.rst

```diff
@@ -0,0 +1,185 @@
+                  Risk('child_growth_failure'),
+                  DirectEffect('child_growth_failure', 'infected_with_diarrhea.incidence_rate'),
+                  DirectEffect('child_growth_failure', 'infected_with_diarrhea.excess_mortality_rate'),
+                  MagicWandIntervention('breastfeeding_promotion', 'child_growth_failure.proportion_exposed'),]
+
+Configurations
+~~~~~~~~~~~~~~
+We also need to create a dictionary of the configurations for the components.
+
+.. code-block:: python
+
+    config = {
+       'randomness': {
+           'key_columns': ['entrance_time', 'age'],
+       },
+       'population': {
+           'population_size': 10_000,
+       },
+       'diarrhea': {
+           'incidence': 2.5,        # Approximately 2.5 cases per person per year.
+           'remission': 42,         # Approximately 6 day median recovery time
+           'excess_mortality': 12,  # Approximately 22 % of cases result in death
+       },
+       'child_growth_failure': {
+           'proportion_exposed': 0.5,
+       },
+       'effect_of_child_growth_failure_on_infected_with_diarrhea.incidence_rate': {
+           'relative_risk': 5,
+       },
+       'effect_of_child_growth_failure_on_infected_with_diarrhea.excess_mortality_rate': {
+           'relative_risk': 5,
+       },
+       'breastfeeding_promotion': {
+           'effect_size': 0.5,
+       },
+    }
+
+Initialize
+~~~~~~~~~~
+
+We can setup and progress the simulation as we did above. The object we get back, an ``InteractiveContext``, is the same
+no matter which helper function you use, though it may or may not have had its setup function called. Here, we will use
+``setup_simulation`` to automatically initialize and setup our simulation context.
+
+.. code-block:: python
+
+    from vivarium_public_health.interactive import setup_simulation
+
+    sim = setup_simulation(components, config)
+    sim.step()
+
+.. note::
+    If we did not need to alter the configuration we could have used the function's counterpart from ``interactive``
+    that would implicitly call setup for us, ``setup_simulation()``.
+
+We can now progress the simulation as we did above. The object we get back, an ``InteractiveContext``, is the same
+no matter which helper function you use.
+
+Progressing the Simulation
+==========================
+
+A simulation can be progressed in several ways, either in terms of steps or in terms of time in the context of the
+simulation. Above, we advanced the simulation by one step using ``step()``, which is a unit of time dictated by the
+model's configuration.  We could also take an arbitrary amount of steps using ``takesteps()``.  Perhaps the simplest
+way is to call ``run()``, which will run the simulation from it's specified start and end times.
+
+A simulation can also be run in terms of its timeframe. ``run_for()`` and ``run_until()`` will run for a period of time
+or until a timestamp, but the simulation's time frame is needed to make sense of this.
```

Comment:
the last part of this sentence is a little confusing re the "simulation's time frame" since that's not really a phrase we've used before. I think what you're saying is we need to know the step size and when the sim starts and ends to use these correctly?

Reply:
yeah I haven't figured out the right way to phrase this

Reply:
I'm just going to expand this with some examples, too. I don't see a way around it